### ISSUE TITLE
Ver 2.6.0 - Web Portal fails to install

### ISSUE THREAD

**Issue by abcd567a** [jprochazka/adsb-receiver#354]
pi@raspberrypi:~/adsb-receiver $ ./install.sh
..........................
..........................
The ADS-B Receiver Project v2.6.0 Installer

  Setting up the ADS-B Receiver Project Portal...
  ------------------------------------------------------------------------------

  Installing packages needed to fulfill dependencies...

  Checking if the package lighttpd is installed... [OK]
  Checking if the package php-cgi is installed... [NOT INSTALLED]
  Installing the package php-cgi...

Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package php-cgi
..........................
..........................
..........................
  Attempting to Install the package php-cgi again in 5 seconds (ATTEMPT 5 OF 5)...

Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package php-cgi

  INSTALLATION HALTED!
  UNABLE TO INSTALL A REQUIRED PACKAGE.
  SETUP HAS BEEN TERMINATED!

The package "php-cgi" could not be installed in 5 attempts.

    ANY FURTHER SETUP AND/OR INSTALLATION REQUESTS HAVE BEEN TERMINIATED

  Cleaning up log file...
pi@raspberrypi:~/adsb-receiver $ 

===================================

SUCCEEDED AFTER WORKAROUND:

sudo nano ~/adsb-receiver/bash/portal/install.sh
Made necessary changes as shown in screenshot, then proceed with installation
cd ~/adsb-receiver
./install.sh
SUCCESS

![web portal workaround](https://user-images.githubusercontent.com/28452511/31314638-cded13fc-abf4-11e7-93ad-e9d51dca14ab.png)

**Comment by abcd567a** [comment]
[BUG FOUND]
---------------
On **Raspberry Pi (armhf)** running Raspbian Jessie or Stretch, and on **Desktop (amd64/i386)** running Pixel x86, the Web Portal install fails due to script trying to install wrong packages as follows:
php-cgi instead of php5-cgi
php-json instead of php5-json
php-xml instead of php5-xml 

I have finally found 2 bugs are causing this.

## BUG 1 of 2: 
The version numbers are not added to package names. This is cause by wrong type of quotes in lines 240 &  241. When I replaced the quotes by correct type, the script added version number 5 to package names and installed php5-cgi and php5-json. 

![wrong type quotes](https://user-images.githubusercontent.com/28452511/31321590-62fe93c4-ac56-11e7-8c03-5d361d31b9cf.png)
.
.
.
## BUG 2 of 2: 
After fixing Bug-1, when I run the install.sh, it installed php5-cgi and php5-json, but when it reached the stage to install php5-xml, it halted saying package not available. I tried to install php5-xml manually, and got message copied below.

```
pi@piaware:~$ sudo apt-get install php5-xml
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package php5-xml

```
I then commented-out the line 281, and tried again and install succeeded.
`281  #  CheckPackage php${DISTRO_PHP_VERSION}-xml`
.
.
**NOTE:**
 Later I checked Debian Wiki and found php5-xml package is not available, but php7.0-xml package is available in Debian repositories.

**Comment by abcd567a** [comment]
## FURTHER INVESTIGATION

On my Ras Pi, I created a test script, copying the relevant part **(lines 239 to 253)** from file bash/portal/install.sh to a blank file "test.sh". After last line (line 281), I added following 3 lines to print the outputs:

```
echo "DISTRO ID = "$DISTRO_ID
echo "Distro Release = "$DISTRO_RELEASE
echo "Distro PHP Version = "$DISTRO_PHP_VERSION
```

I ran the file "test.sh", and found that, contrary to my conclusion in previous post, the back-quotes ( `   ` ) in lines 240 and 241 are correct, and these lines and give proper output. 

However variable DISTRO_PHP_VERSION has null value instead of 7.0 or 5. The blank value of php version results in installation script trying to install php-cgi instead of php5-cgi, and upon failure to find php-cgi, halt and quit further execution of installation script.

```
#created blank file test.sh, and copy-pasted code above
pi@raspberrypi:~ $ sudo nano test.sh

#run the file test.sh
pi@raspberrypi:~ $ sudo chmod +x test.sh
pi@raspberrypi:~ $ ./test.sh
DISTRO ID = debian
Distro Release = 8
Distro PHP Version = 

```
I ran the "test.sh" a second time after replacing back-quotes ( `   ` ) in lines 240 and 241 by straight-quotes ( '   ' ). The code in lines 240 & 241 failed, and gave wrong outputs. This failure resulted in variable DISTRO_PHP_VERSION to have its default value of 5, and prevented installation script from halting and quitting. 

```
#After replacing back-quotes by straight-quotes
pi@raspberrypi:~ $ ./test.sh
DISTRO ID = . /etc/os-release; echo ${ID/*, /}
Distro Release = . /etc/os-release; echo ${VERSION_ID/*, /}
Distro PHP Version = 5
```
### Oliver Jowett (obj) has responded to my above tests in [Flightaware Forums](https://discussions.flightaware.com/t/ads-b-receiver-project-setup-scripts/17807/1657) as follows:


> `if [[ $DISTRO_RELEASE -ge "9" ]]; then DISTRO_PHP_VERSION="7.0"; fi`
> 
> This either needs an `else` clause or you need to set DISTRO_PHP_VERSION to the default value (5) before the `case`
> 
.
.
.
.

**Here is the code of file "test.php"**

```
# Detect the OS distribution and version.
DISTRO_ID=`. /etc/os-release; echo ${ID/*, /}`
DISTRO_RELEASE=`. /etc/os-release; echo ${VERSION_ID/*, /}`

case $DISTRO_ID in
    debian|raspbian)
        if [[ $DISTRO_RELEASE -ge "9" ]]; then DISTRO_PHP_VERSION="7.0"; fi
        ;;
    ubuntu)
        if [[ $DISTRO_RELEASE -ge "16.04" ]]; then DISTRO_PHP_VERSION="7.0"; fi
        ;;
    *)
        DISTRO_PHP_VERSION="5"
        ;;
esac

#Following 3 lines added by me to check outputs
echo "DISTRO ID = "$DISTRO_ID
echo "Distro Release = "$DISTRO_RELEASE
echo "Distro PHP Version = "$DISTRO_PHP_VERSION
```

**Comment by abcd567a** [comment]
Tested “test.sh” on 3 distros (Raspbian Pixel x86, Ubuntu 16.04 amd64, Debian 9 amd64)

Found it fails on Raspbian Pixel x86 and Ubuntu 16.04 amd64, but performs OK on Debian 9 amd64. 
.

**Raspbian Pixel x86 (gives same results on Rasberry Pi/Raspbian Jessie and Stretch)**
![snapshot1](https://user-images.githubusercontent.com/28452511/31392225-3f17ff04-adc8-11e7-8fe7-2240c788befe.png)
.
.

**Ubuntu 16.04 amd64**
![screenshot from 2017-10-10 08-49-14](https://user-images.githubusercontent.com/28452511/31392260-528fa76c-adc8-11e7-98a9-05b19bdb00e8.png)
.
.
.

**Debian 9 amd64**
![screenshot_20171010_134901](https://user-images.githubusercontent.com/28452511/31392278-614ffbee-adc8-11e7-9e8f-d27b27b62494.png)

**Comment by jprochazka** [comment]
Thanks for taking a look at what's been made available so far. Input from the community always helps immensely when getting to the tail end of things before a release. Actually the problem was not assigning a default value for the DISTRO_PHP_VERSION value. This value was not being set in the case statement identifying the OS and version. This should be fixed now in commit c5ccb8850dbcd4001bdcfab37bc347d56c74e521.
